pkg/querier/blocks_store_queryable.go

```diff
@@ -595,6 +602,10 @@ func (q *blocksStoreQuerier) fetchSeriesFromStores(
 
 			stream, err := c.Series(gCtx, req)
 			if err != nil {
+				if isRetryableError(err) {
+					level.Warn(spanLog).Log("err", errors.Wrapf(err, "failed to fetch series from %s", c.RemoteAddress()))
```

Comment:
Can we make it clear that this is a retriable error in the log message? Something like "failed to fetch series from %s due to retriable error".